### ISSUE TITLE
🐛 parse: Parse error in tailwind @variant syntax

### ISSUE THREAD

**Issue by jbergstroem** [biomejs/biome#8269]
### Environment information

```block
CLI:
  Version:                      2.3.7
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-ghostty
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         /path/to/repo/biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

..using:

```json
{
  "css": {
    "parser": {
      "tailwindDirectives": true
    }
  }
}
```

### What happened?

Biome complains about parsing the following syntax which tailwind has no problems consuming:
```css
@utility container {
  padding-inline: 1rem;
  margin-inline: auto;

  @variant 2xl {
    max-width: 1400px;
  }
}
```

```bash
preset.css:72:13 lint/correctness/noUnknownTypeSelector ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Unknown type selector is not allowed.

    70 │   margin-inline: auto;
    71 │
  > 72 │   @variant 2xl {
       │             ^^
    73 │     max-width: 1400px;
    74 │   }

  ℹ See MDN web docs for more details.

  ℹ Consider replacing the unknown type selector with valid one.
```

I'm guessing it fails because css variables cannot start with a digit (if I remember correctly) - but this seems to be pretty common: https://github.com/search?type=code&q=%22%40variant+2xl+%7B%22

### Expected result

Biome should accept this syntax given its widespread use and acceptance by the tailwind parser.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by eenlars** [comment]
Just wanted to bump this — we're hitting this in our monorepo as well (biome 2.3.13, Tailwind CSS 4).

Even with `tailwindDirectives: true` in our biome config, the `@variant dark (.dark &)` syntax causes parse errors that fail `biome ci`. The `@apply` and `@theme` directives work fine with the flag, but `@variant` does not.

We've had to work around this by disabling all CSS processing in our biome config via an override:

```json
{
  "overrides": [
    {
      "includes": ["**/*.css"],
      "linter": { "enabled": false },
      "formatter": { "enabled": false },
      "assist": { "enabled": false }
    }
  ]
}
```

This effectively means we can't use biome for CSS at all, which is unfortunate. Would love to see `@variant` support added to the `tailwindDirectives` parser option.

**Comment by dyc3** [comment]
@eenlars that looks like a separate bug. Can you post a new issue for it? Please include a link to tailwind docs or test cases that show the syntax is valid.

To be clear, `@variant` is supposed to be supported, but we don't have a clear reference for the syntax so it's a bit hard to catch all these cases.
